store/sqlstore/post_store.go

```diff
@@ -1608,7 +1617,6 @@ func (s *SqlPostStore) Search(teamId string, userId string, params *model.Search
 
 func (s *SqlPostStore) search(teamId string, userId string, params *model.SearchParams, channelsByName bool, userByUsername bool) (*model.PostList, error) {
```

Comment:
Can we add some tests to verify the new functionality to return posts from all teams if `teamID` is not set?

Reply:
Yes, so that's at the API level. We need tests at the store level too to pin down any new change in functionality. The API layer and store layer work at different levels and cover different sections of code. So it's good to have everything covered.

Reply:
There is one test added to `api4/post_test.go` and a test added to `store/search_test/post_layer.go`. Where I haven't added any test is in `app/post_test.go` (not sure if needed). Still not sure where exactly do you want the new test to be. :sweat_smile: